mm2src/coins/nft.rs

```diff
@@ -205,29 +205,31 @@ async fn get_moralis_nft_list(ctx: &MmArc, chain: &Chain, url: &Url) -> MmResult
         if let Some(nfts_list) = response["result"].as_array() {
             for nft_json in nfts_list {
                 let nft_wrapper: NftWrapper = serde_json::from_str(&nft_json.to_string())?;
-                let uri_meta = try_get_uri_meta(&nft_wrapper.token_uri).await?;
-                let nft = Nft {
-                    chain: *chain,
-                    token_address: nft_wrapper.token_address,
-                    token_id: nft_wrapper.token_id.0,
-                    amount: nft_wrapper.amount.0,
-                    owner_of: nft_wrapper.owner_of,
-                    token_hash: nft_wrapper.token_hash,
-                    block_number_minted: *nft_wrapper.block_number_minted,
-                    block_number: *nft_wrapper.block_number,
-                    contract_type: nft_wrapper.contract_type.0,
-                    collection_name: nft_wrapper.name,
-                    symbol: nft_wrapper.symbol,
-                    token_uri: nft_wrapper.token_uri,
-                    metadata: nft_wrapper.metadata,
-                    last_token_uri_sync: nft_wrapper.last_token_uri_sync,
-                    last_metadata_sync: nft_wrapper.last_metadata_sync,
-                    minter_address: nft_wrapper.minter_address,
-                    possible_spam: nft_wrapper.possible_spam,
-                    uri_meta,
-                };
-                // collect NFTs from the page
-                res_list.push(nft);
+                if nft_wrapper.contract_type.is_some() {
```

Comment:
Why not use if `let Some(contract_type) = nft_wrapper.contract_type`, it can even be done better the below way to avoid nesting
```rust
let contract_type = match nft_wrapper.contract_type {
    Some(contract_type) => contract_type.0,
    None => continue,
};
```

Reply:
Done
I found myself thinking that maybe we just have different tastes :D First time I wanted to use if let Some(). I guess I still haven't gotten rid of the habit of checking through `if`.

Reply:
> I found myself thinking that maybe we just have different tastes :D

The main idea is to avoid using unwrap since the `is_some()` check can be changed carelessly in the future by anyone without noticing it in review and then the `unwrap` will lead to panicking. The other changes come down to tastes as you mentioned :) For me I like to reduce the nesting as much as possible in the code, it looks cleaner for me that way. You can check [flat coding](https://dev.to/recursivefaults/the-life-changing-magic-of-flat-code-h0m) and [guard clauses](https://betterprogramming.pub/refactoring-guard-clauses-2ceeaa1a9da) if you want to know more about why this is considered better by many devs.